### PR TITLE
chore: remove legacy API docs

### LINKED ISSUE #3058: Consider moving `ApiDocs` outside of the documentation.
### Summary

Currently our API docs are appended to our handcrafted documentation ([example](https://docs.fuel.network/docs/fuels-ts/api/)). This has been the source of a couple of issues and it is aesthetically unappealing in the docs-hub. We should explore the possibility of hosting this ourself (via Vercel perhaps), and adding a reference to this (like the Rust SDK does [here](https://docs.fuel.network/docs/fuels-rs/reference/)).

- [ ] Deploy `ApiDocs`
- [x] Add reference within our docs to this new deployment
- [x] Remove all legacy resources that enabled this process
- [ ] Ensure compatibility with the `docs-hub`

### PR DESCRIPTION
**Pull request by danielbate**
- Closes #3058

# Release notes

In this release, we:

- Shipped a complete version of the [API reference](https://fuels-ts-docs-api.vercel.app/)

# Summary

Removes the legacy API docs and moves links over to the isolated version.

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: apps/docs/scripts/typedoc-postbuild.ts
Comment (petertonysmith94) [inline apps/docs/scripts/typedoc-postbuild.ts:1]: 🔥

File: .github/workflows/release.yaml
Comment (petertonysmith94) [inline .github/workflows/release.yaml:1]: Can we not delete steps `Checkout API Docs` and `Generate and Push API Docs`?

https://github.com/FuelLabs/fuels-ts/pull/3416/files#diff-e426ed45842837026e10e66af23d9c7077e89eacbe6958ce7cb991130ad05adaR125-R141

Comment (danielbate) [reply .github/workflows/release.yaml:1]: https://github.com/FuelLabs/fuels-ts/pull/3416/commits/fd9ac01ffbe9630886ba55c0e8a03d8993528da1

File: .changeset/calm-buttons-help.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+---
+
+chore: remove legacy API docs
```
Comment (danielbate) [inline .changeset/calm-buttons-help.md:4]: > [!WARNING]
> This PR is ready but requires [this change in docs hub](https://github.com/FuelLabs/docs-hub/pull/485) to be merged before merging this.

**Review by arboleya — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Pull request merged**